extensions/jms/src/main/java/org/mule/extensions/jms/api/source/JmsListener.java

```diff
@@ -190,47 +202,58 @@ public void onStart(SourceCallback<Object, JmsAttributes> sourceCallback) throws
   @Override
   public void onStop() {
     if (LOGGER.isDebugEnabled()) {
-      LOGGER.debug("Stopping JMSSubscriber source");
+      LOGGER.debug("Stopping JMS Message Listener");
     }
 
-    jmsLock.unlock();
-
-    try {
-      if (ackMode.equals(AUTO) || ackMode.equals(DUPS_OK) || ackMode.equals(MANUAL)) {
-        session.get().recover();
+    if (synchronous) {
```

Comment:
why do you still need this synchronous thing? Isn't this achieved by simply setting only one consumer?